File: web/hospital-web/src/api/medicamentos.ts
import { http } from "./httpClient";

// Interfaz exacta al DTO de C#
export interface MedicamentoDto {
  idMedicamento: number;
  descripcion: string;
  tipo: string;
  capacidad: string;
  precio: number;
  stock: number;
  caducidad: string; // ISO Date String
}

// GET: /api/Medicamentos
export async function listarMedicamentos() {
  const res = await http.get<MedicamentoDto[]>("/api/Medicamentos");
  return res.data;
}

// PUT: /api/Medicamentos/{id}/stock
export async function actualizarStockMedicamento(id: number, nuevoStock: number) {
  const res = await http.put(`/api/Medicamentos/${id}/stock`, {
    stock: nuevoStock
  });
  return res.data;
}

File: api/Hospital.Api/Dtos/Farmacia/Medicamentos.cs
using System;

namespace Hospital.Api.Dtos.Farmacia;

public class MedicamentoDto
{
    public int IdMedicamento { get; set; }
    public string Descripcion { get; set; } = string.Empty; // Inicializado para evitar warning
    public string Tipo { get; set; } = string.Empty;
    public string Capacidad { get; set; } = string.Empty;
    public decimal Precio { get; set; }
    public int Stock { get; set; }
    public DateOnly Caducidad { get; set; } // <--- CAMBIO: DateOnly para coincidir con tu BD
}

public class UpdateStockDto
{
    public int Stock { get; set; }
}